rules/opentitan/defs.bzl

```diff
@@ -74,6 +75,17 @@ DEFAULT_TEST_FAILURE_MSG = "({})|({})".format(
     ROM_BOOT_FAILURE_MSG,
 )
 
+def rsa_key_for_lc_state(key_structs, hw_lc_state):
+    """Return a dictionary containing a single that can be used in the given
+    LC state. The format of the dictionary is compatible with opentitan_test.
+    """
+    keys = [k for k in key_structs if (not k.rsa or key_allowed_in_lc_state(k.rsa, hw_lc_state))]
+    if len(keys) == 0:
+        fail("There are no RSA keys compatible with HW LC state {} in key structs".format(hw_lc_state))
+    return {
+        keys[0].rsa.label: keys[0].rsa.label
+    }
+
```

Comment:
Lets move this to another file (suggestion: `rules/opentitan/keyutil.bzl`) and then re-export it in `defs.bzl`.  I want to keep `defs.bzl` as the public interface and put private implementations in other files.

The reason you're seeing a failure in CI is the returned dictionary in this function has in incorrect value.  The value is currently the label of the key, but it should be just the key's nickname (e.g. `test_key_0`).  We could add some error checking to `rules/signing.bzl%key_from_dict` to detect this:

```
def key_ext(rsa, spx):
  ...
  key, name = key.items()[0]
  if '/' in name or '.' in name or ':' in name:
    fail("Invalid key nickname for ", str(key), ".  Nickname ", name, " is invalid.")
  ...
```

 The detailed reason this is causing an error is that the label looks like a pathname to the filename/pathname manipulations in `signign.bzl` and that causes an incorrect output filename to be computed.